judge/models/interface.py

```diff
@@ -61,6 +64,11 @@ def pattern(self, cache={}):
             pattern = cache[self.regex] = re.compile(self.regex, re.VERBOSE)
             return pattern
 
+    @staticmethod
+    def for_user(user):
+        all_navbar_items = NavigationBar.objects.all()
+        return list(item for item in all_navbar_items if item.permission is None or user.has_perm(item.permission))
```

Comment:
Why are you using `list(...)` to emulate list comprehensions when you can use the real thing?